kmultiversx/src/kmultiversx/kdist/mx-semantics/esdt.md

```diff
@@ -90,6 +130,65 @@ module ESDT
         </account>
       [priority(61)]
 
+    rule [addToESDTBalance-new-err]:
+        <commands> addToESDTBalance(_ACCT, _TOKEN, _DELTA, false)
+                => #throwExceptionBs(ExecutionFailed, b"new NFT data on sender") 
+                   ...
+        </commands>
+      [priority(61)]
+
+```
+
+- Add NFT to an account
+
+```k
+    syntax InternalCmd ::= addNFTToDestination(Bytes, Bytes, Bytes, Int)
+ // --------------------------------------------------------------------
+    rule [addNFTToDestination-existing]:
+        <commands> addNFTToDestination(FROM, TO, TOKEN, DELTA) => .K ... </commands>
+        <account>
+          <address> FROM </address>
+          <esdtData>
+            <esdtId> TOKEN </esdtId>
+            <esdtBalance> ORIGFROM => ORIGFROM -Int DELTA </esdtBalance>
+            ...
+          </esdtData>
+          ...
+        </account>
+        <account>
+          <address> TO </address>
+          <esdtData>
+            <esdtId> TOKEN </esdtId>
+            <esdtBalance> ORIGFROM => ORIGFROM +Int DELTA </esdtBalance>
```

Comment:
s/ORIGFROM/ORIGTO/?

Also, what happens if FROM == TO?

Also, wouldn't it be simpler if this was split into an "removeFromSource" and "addToDestination"? The way it is written right now, the operation is called "addNFTToDestination", so its name implies either a mint operation, or the second half of a transfer, but it's actually a full transfer.